schema/shared/src/main/scala/zio/blocks/schema/SchemaAspect.scala

```diff
@@ -0,0 +1,24 @@
+package zio.blocks.schema
+
+trait SchemaAspect[-Min, +Max] {
+
+  def apply[A >: Max <: Min](reflect: Reflect.Bound[A]): Reflect.Bound[A]
```

Comment:
The other method I would **love** to have here is `def recursive(implicit ev1: Min >:> Any, ev2: Max <:< Nothing): SchemaAspect[Min, Max]`.

By default this would return the same thing, but subclasses could override to provide recursive behavior.

For example, the `doc` aspect could override to provide a version that attaches the docs to all subtrees, not just the top one (note: doc does not make sense to have a recursive version, so I would not actually do that, but something like `kebabCase` makes sense to have in both flat and recursive variants, so you can do `foo @@ kebabCase` but also `foo @@ kebabCase.recursive`).